src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/SpaceBookingController.kt

```diff
@@ -120,7 +127,22 @@ class SpaceBookingController(
     bookingId: UUID,
     cas1NewArrival: Cas1NewArrival,
   ): ResponseEntity<Unit> {
-    return super.premisesPremisesIdSpaceBookingsBookingIdArrivalPost(premisesId, bookingId, cas1NewArrival)
+    userAccessService.ensureCurrentUserHasPermission(UserPermission.CAS1_SPACE_BOOKING_RECORD_ARRIVAL)
+
+    val result = cas1SpaceBookingService.recordArrivalForBooking(
+      premisesId,
+      bookingId,
+      cas1NewArrival,
+    )
+
+    when (result) {
+      is CasResult.Success -> Unit
+      is CasResult.FieldValidationError -> throw BadRequestProblem(invalidParams = result.validationMessages)
+      is CasResult.ConflictError -> throw ConflictProblem(id = result.conflictingEntityId, conflictReason = result.message)
+      else -> throw InternalServerErrorProblem("An error occurred recording an arrival at premises: $premisesId for space booking: $bookingId")
+    }
```

Comment:
I think we can just use extractEntityFromCasResult() to deal with error cases, see an example in getSpaceBookingById